src/service/video-manager-impl.js

```diff
@@ -436,6 +540,136 @@ class VideoEntry {
   }
 
   /**
+   * Called when the video's position in the viewport changed.
+   * @param {PositionInViewportEntryDef} newPos
+   * @private
+   */
+  onPositionChanged_(newPos) {
+
+    const vidElement = scopedQuerySelector(this.video.element, 'video, iframe');
+    const vidStyle = vidElement.style;
+
+    // How much to scale the video by when minimized
+    const SCALE = 0.6;
+
+    // Temporary fix until PositionObserver somehow tracks objects outside of
+    // the viewport
+    if (!newPos.positionRect
+       && this.getPlayingState() == PlayingStates.PLAYING_MANUAL
+       && this.minimizePosition_ != MinimizePositions.DEFAULT) {
+      // Hide the controls.
+      this.video.hideControls();
+      vidElement.classList.add('i-amphtml-dockable-video-minimizing');
+      vidStyle.height = SCALE * this.initialHeight_ + 'px';
+      vidStyle.width = SCALE * this.initialWidth_ + 'px';
+      vidStyle.borderRadius = '6px';
```

Comment:
I don't think it's possible to move this to css, because in the case where say we want the minimized video to be 50% of the original one and the position observer is triggered only on 2 animations frames (the user scrolls to fast to the top), on the first `onPositionChanged` call, the animation would have scaled the element by 25% (sets `style="transform:scale(0.75)"` and thus overrides the CSS value) and on the second call since we only rely on CSS, the scale would stay at 0.75 instead of going to 0.5. This is mainly because PositionObserver doesn't return the position of the element when it goes out of the view.